dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -222,59 +225,33 @@ void FramesDecoderGpu::SeekFrame(int frame_id) {
   FramesDecoder::SeekFrame(frame_id);
 }
 
-bool FramesDecoderGpu::ReadNextFrame(uint8_t *data, bool copy_to_output) {
-  // No more frames in the file
-  if (next_frame_idx_ == -1) {
-    return false;
-  }
-
+bool FramesDecoderGpu::ReadNextFrameWithIndex(uint8_t *data, bool copy_to_output) {
   // Check if requested frame was buffered earlier
   for (auto &frame : frame_buffer_) {
-    if (frame.pts_ == Index(next_frame_idx_).pts) {
-      if (copy_to_output) {
-        copyD2D(data, frame.frame_.data(), FrameSize());
+    if (frame.pts_ != -1) {
+      if (frame.pts_ == Index(next_frame_idx_).pts) {
```

Comment:
nitpick: Wouldn't it be slightly more readable with just `&&`?

Btw. Is this enforced by the changes in this PR or more of a "hardnening" the check unrealted to the changes?

Reply:
Fixed